src/V3EmitV.cpp

```diff
@@ -629,11 +629,30 @@ class EmitVBaseVisitor VL_NOT_FINAL : public EmitCBaseVisitor {
     virtual void visit(AstTopScope* nodep) override { iterateChildren(nodep); }
     virtual void visit(AstScope* nodep) override { iterateChildren(nodep); }
     virtual void visit(AstVar* nodep) override {
-        putfs(nodep, nodep->verilogKwd());
-        puts(" ");
-        iterate(nodep->dtypep());
-        puts(" ");
-        puts(nodep->prettyName());
+        if (nodep->isIO()) {
+            putfs(nodep, nodep->verilogKwd());
+            puts(" ");
+        }
+        std::vector<AstUnpackArrayDType*> unpackps;
+        for (AstNodeDType* dtypep = nodep->dtypep(); dtypep;) {
+            if (AstUnpackArrayDType* unpackp = VN_CAST(dtypep, UnpackArrayDType)) {
```

Comment:
```suggestion
        for (AstNodeDType* dtypep = nodep->dtypep(); dtypep;) {
            dtypep = dtypep->skipRefp();
            if (AstUnpackArrayDType* unpackp = VN_CAST(dtypep, UnpackArrayDType)) {
```

And have your test use.reference through a typedef.